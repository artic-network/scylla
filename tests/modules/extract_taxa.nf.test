nextflow_workflow {

    name "extract_taxa module tests"
    script "modules/extract_all.nf"
    workflow "extract_all"
    tag "unit"
    tag "ci"

    test("test_extract_all_success") {

        when {
            params {
                kreport_splits                   = "Bacteria Viruses Metazoa"
                extract_thresholds               = [
                    'Viruses': [
                        "taxon_rank": "F",
                        "min_reads": 2,
                        "min_percent": 0,
                    ],
                    'Metazoa': [
                        "taxon_rank": "G",
                        "min_reads": 100000000,
                        "min_percent": 100,
                    ],
                    'Bacteria': [
                        "taxon_rank": "G",
                        "min_reads": 500,
                        "min_percent": 1,
                    ],
                    'default': [
                        "taxon_rank": "G",
                        "min_reads": 100,
                        "min_percent": 0.1,
                    ],
                ]
            }
            workflow {
                """
                input[0] = Channel.of(
                    tuple(
                        "extract_test",
                        file("${projectDir}/tests/test-data/extract_all/cchf_mock.fastq", checkIfExists: true)
                    )
                )
                input[1] = Channel.of(
                    tuple(
                        "extract_test",
                        "PlusPF",
                        file("${projectDir}/tests/test-data/extract_all/cchf_kraken_output.txt", checkIfExists: true)
                    )
                )
                input[2] = Channel.of(
                    tuple(
                        "extract_test",
                        "PlsuPF",
                        file("${projectDir}/tests/test-data/extract_all/cchf_kraken_report.txt", checkIfExists: true)
                    )
                )
                input[3] = file("${projectDir}/tests/test-data/hcid/3052518/tax_dump", checkIfExists: true)
                """
            }
        }

        then {
            assert snapshot(workflow.out, path("${outputDir}/extract_test/reads_by_taxa/1980415.fastq.gz")).match()
        }

    }
}
