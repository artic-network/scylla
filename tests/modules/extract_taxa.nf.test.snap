{
    "test_extract_all_success": {
        "content": [
            {
                "0": [
                    [
                        "extract_test",
                        "cchf_kraken_report.json:md5,bef08c8bdf6832b53d11fbf7fbe2f86e"
                    ]
                ],
                "1": [
                    
                ],
                "kraken_json": [
                    [
                        "extract_test",
                        "cchf_kraken_report.json:md5,bef08c8bdf6832b53d11fbf7fbe2f86e"
                    ]
                ],
                "virus": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-15T13:21:41.382000153"
    }
}